README.md

```diff
@@ -1,104 +1,210 @@
+wish to use a different provider, make sure they support Kubernetes and Terraform and follow the suitable documentation
+to configure it.
 
-Check out the [`qhub` documentation][docs] for more detailed information.
+#### 2. Authentication (using OAuth and GitHub)
+In order to use GitHub actions, QHub will request a `client_id` and `client_secret`. To create those, you will need to 
+sign up for a [GitHub Developer account](https://github.com/settings/developers) and follow the steps in the Auth0 page
+on [how to Connect Apps to GitHub](https://auth0.com/docs/connections/social/github#set-up-app-in-github). Take note of
+your GitHub `client_id` and `client_secret`.
 
-### rendering the configuration file
+#### 3. Domain
+Finally, you will need to have a domain name for hosting QHub. The [docs](https://qhub.dev/docs/step-by-step-walkthrough.html#cloudflare),
+describe how register DNS using Cloudflare only, but any other DNS provider can be used.
 
-_we need more information here._
+Once all those have been set, it is time to generate the config file.
 
+### Initialize Configuration
+Configuration files can be created with the command `qhub init` from the source folder, followed by the alias of the
+chosen Cloud provider, such as:
+```
+    qhub init do        # for DigitalOcean 
+```
+```bash
+    qhub init aws       # for AWS cloud provider
+```
 ```bash
-qhub render -c qhub-config.yaml -o ./ --force
+    qhub init gcp       # for Google Cloud Provider 
 ```
 
-_what is this business?_
+The command will prompt for a `project name` and a `domain name`. Once typed, it will output instructions on how to 
+create an OAuth app on GitHub, and finally ask for the GitHub `client_id` and `client_secret` of the just app created.
 
-## `qhub` interfaces
+The CLI steps as illustrated on the image below:
 
-The `qhub` api normalizes with the nuances of configuring interactive data science environments across multiple client providers. The python command line interfaces define an initial environment state that is modified, and its changes are propogated by continuous integration.
+![init_cli_output.png](docs/images/init_cli_output.png)
 
-Each `qhub` cloud provider has different configuration specifications; more details can be found at the following links about the [Digital Ocean], [AWS], and [GCP] configurations.
+This will generate the `qhub-config.yaml` configuration file with the project's general information, and settings for 
+security, infrastructure, computational resource profiles and virtual environments. Triggered by [Github Actions], this
+YAML file will set up the automated deployment and handle the scaling of the data science project.
 
+For more details on how to modify the configuration file, check out the [docs](https://qhub.dev/docs/do/configuration.html).
 
-## Installing `qhub`
+> NOTE: To verify if Terraform is installed on your virtual env type `terraform` on your terminal window. In case 
```

Comment:
I think we should cut this section down, and link to the step-by-step guide. I think it's sufficient to say something like:

Once all the necessary credentials are gathered and set as unix environment variables, QHub can be deployed in three commands in under 20 minutes:
```
qhub init...
qhub render...
qhub deploy...
```

Reply:
I am not sure on this one. Just because I tried using QHub and had to manually install Terraform (the reason why it has been added to it).

Let me sleep on this one for now.